torch/_inductor/lowering.py

```diff
@@ -120,6 +120,8 @@ def decode_dtype(dtype: int):
 def is_integer_type(x):
     if isinstance(x, TensorBox):
         return is_integer_dtype(x.get_dtype()) or is_boolean_dtype(x.get_dtype())
+    elif isinstance(x, sympy.Symbol):
+        return x.is_integer
```

Comment:
Heads up: this can potentially return `None`, which means "unknown" in SymPy. I wonder if we should always return a bool here.

Reply:
Sympy's type inference is generally insufficient:

```
>>> x = sympy.Symbol('x', integer=True)
>>> y = sympy.Symbol('y', integer=True)
>>> repr((x/y).is_integer)
'None'
```

In our universe, we want this to be False, as true division always returns a float.